rest-tests/src/test/groovy/org/hawkular/metrics/rest/CORSITest.groovy

```diff
@@ -59,22 +59,51 @@ class CORSITest extends RESTTest {
     }
 
     @Test
-    void testOptionsWithoutOrigin() {
+    void testOptionsWithBadOrigin() {
         def response = hawkularMetrics.options(path: "gauges/test/data",
             headers: [
-                (ACCESS_CONTROL_REQUEST_METHOD): "GET"
+                (ACCESS_CONTROL_REQUEST_METHOD): "OPTIONS",
+                (ORIGIN): "*"
+            ])
+
+        //Expected a 400 because this is be a pre-flight call that should never reach the resource router because
+        //the request origin is bad
+        assertEquals(400, response.status)
+
+        def testOrigin = System.getProperty('hawkular-metrics.test.origin')
+
+        def wrongSchemeOrigin = testOrigin.replaceAll('http://','https://');
+        response = hawkularMetrics.options(path: "gauges/test/data",
+            headers: [
+                (ACCESS_CONTROL_REQUEST_METHOD): "GET",
+                (ORIGIN): wrongSchemeOrigin
+            ])
+
+        //Expected a 400 because this is be a call that should never reach the resource router because
+        //the request origin is bad
+        assertEquals(400, response.status)
+    }
+
+    @Test
+    void testOptionsWithSubdomainOrigin() {
+        def testOrigin = System.getProperty('hawkular-metrics.test.origin')
```

Comment:
You can make `testOrigin` a static field in this class and re-use it in the other test.